### PR TITLE
[TRITON] FP8 MQA optimizations

### PR DESCRIPTION
**Pull request by cagrikymk**
This PR optimizes the FP8 MQA kernel.

- Handle the rows with fewer masked elements first to reduce the tail effect
- Decide the MFMA shape based on the imput (32x32x16 vs 16x16x32)

### CODE REVIEW

File: aiter/ops/triton/fp8_mqa_logits.py [hunk 1]
```diff
@@ -44,6 +44,11 @@ def fp8_mqa_logits(
     stride_kv_s, stride_kv_d = KV.stride()
     stride_w_s, stride_w_h = weights.stride()
     stride_logits_s, stride_logits_k = logits.stride()
+
+    matrix_instr_nonkdim = 32
+    if seq_len <= 1024:
```
Comment (vgokhale) [inline aiter/ops/triton/fp8_mqa_logits.py:50]: This is empirical? May be add a comment to that effect?

File: op_tests/op_benchmarks/triton/bench_fp8_mqa_logits.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+def calculate_tflops(start_inds, end_inds, num_heads_q, head_dim, time_ms):
+    time_s = time_ms * 1e-3
+    start_inds = start_inds.to("cpu").numpy()
+    end_inds = end_inds.to("cpu").numpy()
+    total_flops = 0.0
+    for i in range(len(start_inds)):
+        start = start_inds[i]
+        end = end_inds[i]
+
```
Comment (vgokhale) [inline op_tests/op_benchmarks/triton/bench_fp8_mqa_logits.py]: Nit: Unnecessary blank line

**Review by vgokhale — DISMISSED** [review]

File: aiter/ops/triton/fp8_mqa_logits.py (hunk 1 above)
Comment (cagrikymk) [reply aiter/ops/triton/fp8_mqa_logits.py:50]: Yes, it is, added the related comment

File: op_tests/op_benchmarks/triton/bench_fp8_mqa_logits.py (hunk 2 above)
Comment (cagrikymk) [reply op_tests/op_benchmarks/triton/bench_fp8_mqa_logits.py]: Done
